include/socgraph.php

```diff
@@ -190,181 +187,166 @@ function poco_load_worker($cid, $uid, $zcid, $url) {
 	);
 
 }
+/**
+ * @brief Sanitize the given gcontact data
+ *
+ * @param array $gcontact array with gcontact data
+ *
+ * Generation:
+ *  0: No definition
+ *  1: Profiles on this server
+ *  2: Contacts of profiles on this server
+ *  3: Contacts of contacts of profiles on this server
+ *  4: ...
+ *
+ */
+function sanitized_gcontact(&$gcontact) {
```

Comment:
Here's my thinking. You can have a `sanitized_gcontact($contact)` that would return a boolean depending on whether `$contact` already is sanitized or not. And you can have a `sanitize_gcontact($contact)` that would go through the `$contact` array and change its values, returning a sanitized copy for future use. But not a function that does both at the same time. For me, it would be like `is_bool($var)` transformed $var into a boolean, returning `false` if it can't for any reason.

It is less confusing if filters and transformers are separated, and yes force-calling by reference is generally a bad thing because it simply is unexpected. Here's all the versions I thought about when writing my initial comment.

Renaming the function:
````php
function sanitize_gcontact($contact) {
    ...

    return $contact;
}

$gcontact = sanitize_gcontact($gcontact);
if ($gcontact !== false) {
    $gcid = update_gcontact($gcontact);
    link_gcontact($gcid, $uid, $cid, $zcid);
}
````
Adding exceptions:
````php
function sanitize_gcontact($contact) {
    if ($gcontact['url'] == "") {
         throw new Exception('URL is empty');
    }

    ...

    return $contact;
}

try {
    $gcontact = sanitize_gcontact($gcontact);
    $gcid = update_gcontact($gcontact);
    link_gcontact($gcid, $uid, $cid, $zcid);
} catch (Exception $e) {
    // Error-handling, logging, etc...
}
````
In a class context:
````php
class GContact {
    ...
    
    function sanitize() {
        if ($this->url == "") {
            throw new Exception('URL is empty');
        }

        ...
        
        if ($this->network == NETWORK_STATUSNET) {
            $this->network = "";
        }
    }    
}

$gcontact = new GContact();

try {
    $gcontact->sanitize();
    $gcontact->update();
    $gcontact->link($uid, $cid, $zcid);
} catch (Exception $e) {
    // Error-handling, logging, etc...
}
````
Enabling method chaining:
In a class context:
````php
class GContact {
    ...
    
    function sanitize() {
        if ($this->url == "") {
            throw new Exception('URL is empty');
        }

        ...
        
        if ($this->network == NETWORK_STATUSNET) {
            $this->network = "";
        }

        return $this;
    }    
}

$gcontact = new GContact();

try {
    $gcontact
        ->sanitize()
        ->update()
        ->link($uid, $cid, $zcid);
} catch (Exception $e) {
    // Error-handling, logging, etc...
}
````